packages/common/src/search/types/IHubSearchResult.ts

```diff
@@ -0,0 +1,110 @@
+import { IHubGeography } from "../..";
+
+/**
+ * Standardized light-weight search result structure, applicable to all
+ * types of search results - users, groups, content, events etc
+ */
+export interface IHubSearchResult {
+  /**
+   * Access level of the backing entity
+   */
+  access: "public" | "shared" | "org" | "private";
+  /**
+   * Identifier - could be a guid or a username
+   */
+  id: string;
+  /**
+   * Hub Family: "dataset" | "map" | "document" | "content"
+   * TODO: Can we create a type for this?
+   */
+  family: string;
+  /**
+   * Actual type
+   * For Item backed results, this will be item.type
+   * Otherwise it will be "Group", "User", "Event" etc
+   */
+  type: string;
+  /**
+   * Name or title
+   */
+  name: string;
+  /**
+   * Owner; Applies to Items and Groups
+   */
+  owner: string;
+  /**
+   * Sanitized summary derived from item.snippet, item.description, group.description
+   */
+  summary: string;
+  /**
+   * Date when the entity was created
+   */
+  createdDate: Date;
+  /**
+   * Source of the creation date as a property path
+   * e.g. `item.created`
+   */
+  createdDateSource: string;
+  /**
+   * Date when the entity was last updated
+   * Depending on the entity, this could be derived
+   * in many different ways
+   */
+  updatedDate?: Date;
+  /**
+   * Source of the updated date as a property path
+   * e.g. `featureService.lastEdit`
+   */
+  updatedDateSource?: string;
+  /**
+   * Fully qualified thumbnail url for items, groups, users
+   * Will not have the token appended. Consuming app needs to
+   */
+  thumbnailUrl: string;
+  /**
+   * Hash of pre-computed urls
+   */
+  urls: {
+    /**
+     * Canononical ArcGIS Online Url for this entity (if applicable)
+     */
+    agoUrl?: string;
+    /**
+     * Relative url, which can be appended to a Site's root Url
+     * e.g. /datasets/3ef8c7a, or /events/june-meeting
+     * This is much easier than passing a Site into the search
+     * system so the full url can be computed
+     */
+    siteRelativeUrl: string;
```

Comment:
This is pretty tight coupling between API and the UI routing.

Reply:
Leaving in for now as the API is returning this ( as `relative`)

Reply:
please update the key to _be_ `relative` - there are Hub.js functions (called by `composeContent()`, that generate this `urls` hash and we should use those actual functions to generate a hash w/ the same actual keys. Let me get you the actual function and return type

Reply:
I'm sorry there's not already an interface for this but `urls` should be: 

```ts
interface IHubContentUrls {
  relative?: string
  portalHome?: string
  portalApi?: string
  portalData?: string
  thumbnail?: string
}
```

see https://github.com/Esri/hub.js/blob/8acb34416ba131d954eeb81c56abcc7febab546b/packages/common/src/content/compose.ts#L627-L630

and 

https://github.com/Esri/hub.js/blob/8acb34416ba131d954eeb81c56abcc7febab546b/packages/common/src/content/compose.ts#L338-L360

Reply:
related to the discussion of adopting [OGC Records](https://github.com/opengeospatial/ogcapi-records) (STAC) this would be

```json
{ "links": [
        {
          "href": "...",
          "rel": "self",
          "type": "application/geo+json"
        },
        {
          "href": "...",
          "rel": "alternate",
          "type": "text/html"
        },
        {
          "href": "...",
          "rel": "arcgisportal",
          "type": "text/html"
        }
      ]
```